sage-ios/SAGE/SAGE/AddSchoolDirectorTableViewController.swift

```diff
@@ -0,0 +1,132 @@
+            
+            for director in directorArray {
+                let firstName = director.firstName!
+                let firstLetter = String(firstName[firstName.startIndex.advancedBy(0)]).lowercaseString
+                let firstLetterIndex = charArray[firstLetter]
+                self.directors![firstLetterIndex!].append(director)
+            }
+            
+            self.tableView.reloadData()
+            self.activityIndicator.stopAnimating()
+            self.activityIndicator.hidden = true
+            self.refreshControl?.endRefreshing()
+            
+            }) { (errorMessage) -> Void in
+                self.showErrorAndSetMessage(errorMessage, size: 64.0)
+        }
+    }
+    
+    override func sectionIndexTitlesForTableView(tableView: UITableView) -> [String]? {
+        let alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        var charArray = [String]()
+        for i in 0...25 {
+            let letterChar = alphabet[alphabet.startIndex.advancedBy(i)]
+            let letterString = String(letterChar)
+            charArray.append(letterString)
+        }
+        return charArray
+    }
+    
+    func showErrorAndSetMessage(message: String, size: CGFloat) {
+        let error = self.currentErrorMessage
+        let errorView = super.showError(message, size: size, currentError: error)
+        self.currentErrorMessage = errorView
+    }
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if let directors = self.directors {
+            return directors[section].count
+        } else {
+            return 0
+        }
+    }
+    
+    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 26
+    }
+    
+    override func tableView(tableView: UITableView, sectionForSectionIndexTitle title: String, atIndex index: Int) -> Int {
+        let alphabet = "abcdefghijklmnopqrstuvwxyz"
+        var charArray = [String: Int]()
+        for i in 0...25 {
+            let letterChar = alphabet[alphabet.startIndex.advancedBy(i)]
+            let letterString = String(letterChar)
+            charArray[letterString] = i
+        }
+        return charArray[title.lowercaseString]!
+    }
+    
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let user = self.directors![indexPath.section][indexPath.row]
+        let cell = BrowseMentorsTableViewCell()
+        cell.configureWithUser(user)
+        return cell
+    }
+    
+    override func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return BrowseMentorsTableViewCell.cellHeight()
+    }
+    
+    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let parentVC = self.navigationController!.viewControllers[self.navigationController!.viewControllers.count - 2] as! AddSchoolController
```

Comment:
please dont do this 

if anything make a weak reference to the parentViewController or make a delegate